venues/naacl.org/NAACL/2018/Preprint/process/submissionProcess.js

```diff
@@ -0,0 +1,173 @@
+            } else {
+                return Promise.reject('No notes with the id ' + note.original + ' were found');
+            }
+        })
+        .then(result => done())
+        .catch(error => done(error));
+        return true;
+    }`
+
+    var getBibtex = function(note) {
+      var firstWord = note.content.title.split(' ')[0].toLowerCase();
+
+      return '@unpublished{\
+          \nanonymous2018' + firstWord + ',\
+          \ntitle={' + note.content.title + '},\
+          \nauthor={Anonymous},\
+          \njournal={NAACL 2018 Preprint},\
+          \nyear={2018},\
+          \nnote={anonymous preprint under review}\
+      \n}'
+    };
+
+
+    var addRevisionInvitation = {
+      id: CONF + '/-/Paper' + note.number + '/Add_Revision',
+      signatures: [CONF],
+      writers: [CONF],
+      invitees: note.content.authorids.concat(note.signatures),
+      noninvitees: [],
+      readers: ['everyone'],
+      reply: {
+        forum: note.id,
+        referent: note.id,
+        signatures: invitation.reply.signatures,
+        writers: invitation.reply.writers,
+        readers: invitation.reply.readers,
+        content: invitation.reply.content
+      }
+    }
+
+    var blindSubmission = {
+      original: note.id,
+      invitation: BLIND_SUBMISSION,
+      forum: null,
+      parent: null,
+      signatures: [CONF],
+      writers: [CONF],
+      readers: ['everyone'],
+      content: {
+        authors: ['Anonymous'],
+        authorids: ['Anonymous'],
+        _bibtex: getBibtex(note)
+      }
+    }
+
+
+
+    or3client.or3request(or3client.inviteUrl, addRevisionInvitation, 'POST', token)
+    .then(result => or3client.or3request(or3client.notesUrl, blindSubmission, 'POST', token))
+    .then(savedNote => {
+      //Send an email to the author of the submitted note, confirming its receipt
+      var mail = {
+          "groups": note.content.authorids,
+          "subject": "Confirmation of your submission to NAACL 2018 Preprint: \"" + note.content.title + "\".",
+          "message": `Your submission to NAACL 2018 Preprint has been posted.\n\nTitle: `+note.content.title+`\n\nAbstract: `+note.content.abstract+`\n\nTo view the note, click here: `+baseUrl+`/forum?id=` + savedNote.forum
+      };
+
+      var paperGroup = {
+        id: CONF + '/Paper' + savedNote.number,
+        signatures: [CONF],
+        writers: [CONF],
+        members: [],
+        readers: [CONF],
+        signatories: []
+      };
+      return or3client.or3request(or3client.grpUrl, paperGroup, 'POST', token)
+      .then(savedPaperGroup => {
+
+        var authorGroupId = savedPaperGroup.id + '/Authors';
+        var authorGroup = {
+          id: authorGroupId,
+          signatures: [CONF],
+          writers: [CONF],
+          members: note.content.authorids.concat(note.signatures),
+          readers: [CONF, authorGroupId],
+          signatories: [authorGroupId]
+        };
+
+        var withdrawPaperInvitation = {
```

Comment:
The withdraw process is the same as ICLR?

Reply:
sorry, I missed this comment somehow.

the withdraw process is similar but not exactly the same as ICLR. When an author withdraws a paper here, both the blind note and the original are deleted (ddate is set). This is to prevent the user from having to manually delete both the original and the blind copy separately